src/components/PopupWithForm.js

```diff
@@ -6,19 +6,26 @@ export default class PopupwithForm extends Popup {
     this._handleFormSubmit = handleFormSubmit;
     this._form = this._popupElement.querySelector(".modal__form");
     this._submitBtn = this._popupElement.querySelector(".modal__save");
+    this._inputsArr = Array.from(this._form.querySelectorAll(".modal__input"));
   }
 
   _getInputValues() {
-    const inputsArr = Array.from(this._form.querySelectorAll(".modal__input"));
-
     const inputValues = {};
 
-    inputsArr.forEach((input) => {
+    this._inputsArr.forEach((input) => {
       inputValues[input.name] = input.value;
     });
     return inputValues;
   }
 
+  isLoading(isLoading) {
+    if (isLoading) {
+      this._submitBtn.textContent = "Saving...";
+    } else {
+      this._submitBtn.textContent = "Save";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```